ESP8266/ESP8266.cpp

```diff
@@ -279,28 +287,24 @@ bool ESP8266::close(int id)
 
 void ESP8266::setTimeout(uint32_t timeout_ms)
 {
-    _parser.setTimeout(timeout_ms);
+    _parser->setTimeout(timeout_ms);
 }
 
-bool ESP8266::readable()
-{
-    return _serial.readable();
-}
-
-bool ESP8266::writeable()
-{
-    return _serial.writeable();
+void ESP8266::stream_event_cb(short events) {
+    if ((events & MBED_POLLIN )|| (events & MBED_POLLOUT)) {
+        callback(_callback);
+    }
 }
-
 void ESP8266::attach(Callback<void()> func)
 {
-    _serial.attach(func);
+    _callback = func;
+    _fh->attach(callback(this, &ESP8266::stream_event_cb));
```

Comment:
Do we care about the type of events? The sigio function does not. Can we skip the `ESP8266::stream_event_cb` function?